auth/src/main/java/org/aerogear/auth/impl/UserPrincipalImpl.java

```diff
@@ -119,8 +121,40 @@ UserPrincipalImpl build() {
      */
     @Override
     public boolean hasRole(final IRole role) {
-//        TODO: to be implemented
-//        return roles.containsKey(role.getRoleName());
+      return roles.contains(role);
+    }
+
+    /**
+     * Check if a user has a specified ClientRole based on param of type String to allow
+     * for better usability.
+     * Returns <code>true</code> if the user has the passed in role
+     * @param role String of role to be checked
+     * @return true or false
+     */
+    @Override
+    public boolean hasClientRole(String role){
```

Comment:
If for some reason in the future we will need to have another type of roles, this way we will have to add new methods. I think it would be better to change this method to 
`hasRole(final String roleName, final Type roleType)`
where `roleType` is an enum.